### ISSUE TITLE
[Bug] Replay Index shows 1 error but no errors in Replay Detail

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry#54698]
Replay Index shows 1 error for all of these 'session' replays: https://nugettrends.sentry.io/replays/?project=1266321&statsPeriod=24h

![image](https://github.com/getsentry/sentry/assets/1633368/d57adc05-fa5e-49ea-9424-8371cc03e659)

[Replay Details](https://nugettrends.sentry.io/replays/0eca4eb8c4514391975b61f0688d4588/?project=1266321&query=&referrer=%2Freplays%2F&statsPeriod=24h&yAxis=count%28%29) show 0:

![image](https://github.com/getsentry/sentry/assets/1633368/4a4d8fcf-4254-4eea-9b2d-cec78540bea6)
including the Errors list:
![image](https://github.com/getsentry/sentry/assets/1633368/42a137d8-cda3-48dc-8aea-b33a7f216ed5)

[Project has a lot of events](https://nugettrends.sentry.io/issues/?project=1266321&referrer=sidebar&statsPeriod=24h) `level=info`:

![image](https://github.com/getsentry/sentry/assets/1633368/5ad5a241-b580-48b5-9763-505423d131ea)

**Comment by ryan953** [comment]
Seems like a classic problem with counting:

The replay list page shows `count_errors: 1,`
```json
# https://nugettrends.sentry.io/api/0/organizations/nugettrends/replays/?field=activity&field=browser&field=count_dead_clicks&field=count_errors&field=count_rage_clicks&field=duration&field=finished_at&field=id&field=is_archived&field=os&field=project_id&field=started_at&field=urls&field=user&per_page=50&project=1266321&query=&sort=-started_at&statsPeriod=24h
{
data: [
{
activity: 4,
browser: {
name: "Edge",
version: "115.0.1901"
},
count_dead_clicks: 0,
count_errors: 1,
count_rage_clicks: 0,
duration: 23,
finished_at: "2023-08-16T15:55:08+00:00",
id: "3b69e86b545a4641a11e3afba2d7c264",
... snip ...
```

The details page also shows the same (it's the same data set):
```json
{
data: {
id: "3b69e86b545a4641a11e3afba2d7c264",
project_id: "1266321",
trace_ids: [ ],
error_ids: [
"ab2e8a58ae6241f8b24fcd7d2a5e6533"
],
... snip ...
count_errors: 1,
... snip ...
```

But that 'error' is actually a "Bootstrap success" info event: https://nugettrends.sentry.io/discover/spa:ab2e8a58ae6241f8b24fcd7d2a5e6533/?field=title&field=event.type&field=project&field=user.display&field=timestamp&field=replayId&homepage=true&name=All+Events&query=id%3Aab2e8a58ae6241f8b24fcd7d2a5e6533&sort=-timestamp&statsPeriod=14d&yAxis=count%28%29

**Comment by ryan953** [comment]
I'd say because it's an "info" event it shouldn't be counted as an error at all. Which means at replay-ingest time we need to know more about the event itself, or drop the `count_errors` field, and use that correct count query and all that jazz.

**Comment by bruno-garcia** [comment]
> We have the same issue(?). We have disabled sampling for non errors in the config (support confirmed that the config is ok) but we have a lot of non error session replays recorded.
> 
> For example: ![Screenshot 2023-11-14 at 15 06 33](https://user-images.githubusercontent.com/19626542/282800091-5f7c34d8-e937-46d9-84fd-a2efad070d91.png) ![Screenshot 2023-11-14 at 15 06 42](https://user-images.githubusercontent.com/19626542/282800120-f6702b0b-7c56-4d77-9c7b-c4376e13d3d6.png)
> 
> For us this is kinda bad as it eats up our session replay quota very fast and we maybe miss real errors. Are those errors real errors and just not shown in the detail view or is there just an issue with counting on the overview page. Because we should normally (as we have disabled it) not record any non error replays.
> 
> Is there any workaround in the meantime?

The workaround for now is to drop the errors client side. By any chance this is a NextJS app with hydration errors? Sentry is now dropping these server-side automatically, but at that point the replay already made it in. We've seen customers with millions of those errors all discarded after ingestion by Sentry.

Looks like this under the )_Inbound Filter_ settings page:
![image](https://github.com/getsentry/sentry/assets/1633368/8ea0e029-d33b-4811-9dbd-53c46f15836b)

**Comment by pinkeyAT** [comment]
> The workaround for now is to drop the errors client side. By any chance this is a NextJS app with hydration errors? Sentry is now dropping these server-side automatically, but at that point the replay already made it in. We've seen customers with millions of those errors all discarded after ingesti ...

Yes you are right, its a NextJS App with a custom express server. As far as a i know we should not have hydration errors and the option from your screenshot is already enabled.

**Comment by bruno-garcia** [comment]
> > The workaround for now is to drop the errors client side. By any chance this is a NextJS app with hydration errors? Sentry is now dropping these server-side automatically, but at that point the replay already made it in. We've seen customers with millions of those errors all discarded after ingesti ...
> 
> Yes you are right, its a NextJS App with a custom express server. As far as a i know we should not have hydration errors and the option from your screenshot is already enabled.

Could you disable that option to see if errors start to show up linked to the replays?

If you prefer not to (those errors will take from your quota) you could pass a link to your org and I could see if there's data getting dropped instead.

**Comment by bruno-garcia** [comment]
@JoshFerge Resolved through https://github.com/getsentry/sentry/pull/59378 ?

**Comment by JoshFerge** [comment]
> @JoshFerge Resolved through #59378 ?

should be!
